cmd/manila-csi-plugin/main.go

```diff
@@ -17,38 +17,87 @@ limitations under the License.
 package main
 
 import (
-	goflag "flag"
-	"k8s.io/cloud-provider-openstack/pkg/csi/manila/manilaclient"
+	"flag"
+	"fmt"
+	"os"
 
-	flag "github.com/spf13/pflag"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 	"k8s.io/cloud-provider-openstack/pkg/csi/manila"
+	"k8s.io/cloud-provider-openstack/pkg/csi/manila/manilaclient"
+	"k8s.io/component-base/logs"
 	"k8s.io/klog"
 )
 
 var (
-	endpoint      = flag.String("endpoint", "unix://tmp/csi.sock", "CSI endpoint")
-	driverName    = flag.String("drivername", "manila.csi.openstack.org", "name of the driver")
-	nodeID        = flag.String("nodeid", "", "this node's ID")
-	protoSelector = flag.String("share-protocol-selector", "", "specifies which Manila share protocol to use")
-	fwdEndpoint   = flag.String("fwdendpoint", "", "CSI Node Plugin endpoint to which all Node Service RPCs are forwarded. Must be able to handle the file-system specified in share-protocol-selector")
+	endpoint      string
+	driverName    string
+	nodeID        string
+	protoSelector string
+	fwdEndpoint   string
+	userAgentData []string
 )
 
 func main() {
-	klog.InitFlags(nil)
-	if err := goflag.Set("logtostderr", "true"); err != nil {
-		klog.Exitf("failed to set logtostderr flag: %v", err)
+	flag.CommandLine.Parse([]string{})
+
+	cmd := &cobra.Command{
+		Use:   os.Args[0],
+		Short: "CSI Manila driver",
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			// Glog requires this otherwise it complains.
+			flag.CommandLine.Parse(nil)
+
+			// This is a temporary hack to enable proper logging until upstream dependencies
+			// are migrated to fully utilize klog instead of glog.
+			klogFlags := flag.NewFlagSet("klog", flag.ExitOnError)
+			klog.InitFlags(klogFlags)
+
+			// Sync the glog and klog flags.
+			cmd.Flags().VisitAll(func(f1 *pflag.Flag) {
+				f2 := klogFlags.Lookup(f1.Name)
+				if f2 != nil {
+					value := f1.Value.String()
+					f2.Value.Set(value)
+				}
+			})
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			d, err := manila.NewDriver(nodeID, driverName, endpoint, fwdEndpoint, protoSelector, &manilaclient.ClientBuilder{UserAgentData: userAgentData})
+			if err != nil {
+				klog.Fatalf("driver initialization failed: %v", err)
+			}
+
+			d.Run()
+		},
 	}
 
-	manilaClientBuilder := &manilaclient.ClientBuilder{}
+	cmd.Flags().AddGoFlagSet(flag.CommandLine)
+
+	cmd.PersistentFlags().StringVar(&endpoint, "endpoint", "unix://tmp/csi.sock", "CSI endpoint")
+	cmd.MarkPersistentFlagRequired("endpoint")
+
+	cmd.PersistentFlags().StringVar(&driverName, "drivername", "manila.csi.openstack.org", "name of the driver")
```

Comment:
shouldn't be there a default name supported here??

Reply:
The name defaults to `manila.csi.openstack.org`...is that ok or do you mean something else?

Reply:
sorry missed it :)